utils.go

```diff
@@ -109,3 +111,33 @@ func newBufConn(conn net.Conn) *bufConn {
 func (bc *bufConn) Read(b []byte) (n int, err error) {
 	return bc.buf.Read(b)
 }
+
+type kVersion struct {
+	versionArr [4]int
+}
+
+func NewVersion(ver string) (*kVersion, error) {
```

Comment:
there is no reason for this to take a string or return an error `NewVersion(0, 9, 0, 1)`  reads cleaner